src/CommitmentsV2/SFA.DAS.CommitmentsV2/Extensions/QueryableApprenticeshipsExtensions.cs

```diff
@@ -178,8 +183,105 @@ private static IQueryable<Apprenticeship> WithAlertsEmployer(this IQueryable<App
+                                 || x.ErrorCode.HasFlag(DataLockErrorCode.Dlock06)) &&
+                                x.TriageStatus == TriageStatus.Unknown &&
+                                !x.IsResolved)
+                        ) ||
+                        (
+                            a.DataLockStatus.Any(x =>
+                                ((int)x.ErrorCode == (int)DataLockErrorCode.Dlock07) &&
+                                x.TriageStatus == TriageStatus.Unknown &&
+                                !x.IsResolved)
+                        )
+                    );
+
+                case Alerts.ChangesPending:
+                    return apprenticeships.Where(a =>
+                        (
+                            a.DataLockStatus.Any(x =>
+                                (
+                                    x.ErrorCode.HasFlag(DataLockErrorCode.Dlock03)
+                                    || x.ErrorCode.HasFlag(DataLockErrorCode.Dlock04)
+                                    || x.ErrorCode.HasFlag(DataLockErrorCode.Dlock05)
+                                    || x.ErrorCode.HasFlag(DataLockErrorCode.Dlock06)
+                                ) &&
+                                x.TriageStatus == TriageStatus.Change &&
+                                !x.IsResolved)
+                        )
+                        ||
+                        (
+                            a.DataLockStatus.Any(x =>
+                                (
+                                    (int)x.ErrorCode == (int)DataLockErrorCode.Dlock07
+                                ) &&
+                                x.TriageStatus == TriageStatus.Change &&
+                                !x.IsResolved)
+                        ));
+                case Alerts.ChangesRequested:
+
+                    if (!isProvider)
+                    {
+                        return apprenticeships.Where(a =>
+                        (
+                            a.DataLockStatus.Any(x =>
+                                x.Status == Status.Fail &&
+                                (x.TriageStatus != TriageStatus.Unknown && x.TriageStatus != TriageStatus.Change) &&
+                                !x.IsResolved)
+                        ));
+                    }
+
+                    return apprenticeships.Where(a =>
+                    (
+                        a.DataLockStatus.Any(x =>
+                        (
+                            x.ErrorCode.HasFlag(DataLockErrorCode.Dlock03)
+                            || x.ErrorCode.HasFlag(DataLockErrorCode.Dlock04)
+                            || x.ErrorCode.HasFlag(DataLockErrorCode.Dlock05)
+                            || x.ErrorCode.HasFlag(DataLockErrorCode.Dlock06)
+
+                        ) && x.TriageStatus == TriageStatus.Restart && !x.IsResolved)
+                    ));
+
+
+                case Alerts.ChangesForReview:
+
+                    if (!isProvider)
```

Comment:
This maybe correct? but if it's not a provider then return records where the Originator is the provider - so you're looking at the records of the other party?